sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -111,3 +112,38 @@ def update(self, signature):
         if not isinstance(signature, six.string_types):
             raise TypeError("The signature used for updating must be a string.")
         self._signature = signature
+
+
+class AzureNamedKeyCredential(object):
+    """Credential type used for working with any service needing a named key that follows patterns
+    established by the other credential types.
+
+    :param str name: The name of the credential used to authenticate to an Azure service.
+    :param str key: The key used to authenticate to an Azure service.
+    :raises: TypeError
+    """
+    def __init__(self, name, key):
+        # type: (str, str) -> None
+        if not isinstance(name, six.string_types) or not isinstance(key, six.string_types):
+            raise TypeError("Both name and key must be Strings.")
+        self._credential = AzureNamedKey(name, key)
+
+    @property
+    def credential(self):
```

Comment:
It feels odd to have a credential property on a credential object. Is this the right name?

Reply:
changed to `named_key`